src/Symfony/Component/Validator/Command/DebugCommand.php

```diff
@@ -0,0 +1,200 @@
+ */
+
+namespace Symfony\Component\Validator\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Mapping\ClassMetadataInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * A console command to debug Validators information.
+ *
+ * @author Loïc Frémont <lc.fremont@gmail.com>
+ */
+class DebugCommand extends Command
+{
+    protected static $defaultName = 'debug:validator';
+
+    private $validator;
+
+    public function __construct(ValidatorInterface $validator)
+    {
+        parent::__construct();
+
+        $this->validator = $validator;
+    }
+
+    protected function configure()
+    {
+        $this
+            ->addArgument('class', InputArgument::OPTIONAL, 'A class to debug')
+            ->addOption('path', null, InputOption::VALUE_OPTIONAL, 'A path for the classes to debug', 'src/Entity')
+            ->addOption('with-results-only', null, InputOption::VALUE_NONE, 'Show only classes with results')
+            ->setDescription('Displays validators for classes')
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> command dumps the validators configuration for all classes in
+`src/Entity` directory.
+
+The <info>%command.name% 'App\Entity\Dummy'</info> command dumps the validators for the dummy class.
+
+The <info>%command.name% --path=src/</info> command dumps the validators for the `src` directory.
+EOF
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $class = $input->getArgument('class');
+        $path = $input->getOption('path');
+
+        if (null !== $class) {
+            $this->dumpValidatorsForClass($input, $output, $class);
+
+            return 0;
+        }
+
+        if (null !== $path) {
+            foreach ($this->getResourcesByPath($path) as $class) {
+                $this->dumpValidatorsForClass($input, $output, $class);
+            }
+
+            return 0;
+        }
+
+        return 0;
+    }
+
+    private function dumpValidatorsForClass(InputInterface $input, OutputInterface $output, string $class): void
+    {
+        $io = new SymfonyStyle($input, $output);
+        $title = sprintf('Information for class "<info>%s</info>"', $class);
+        $rows = [];
+
+        foreach ($this->getConstrainedPropertiesData($class) as $propertyName => $constraintsData) {
+            foreach ($constraintsData as $data) {
+                $rows[] = [$propertyName, $data['class'], implode(', ', $data['groups']), json_encode($data['options'])];
```

Comment:
simple question, why a json encoding?
imho it makes the reading of those sub info more complex than a key pair

what about instead

`message: This value should not be blank. | allowNull: false | foo: bar` ?

Reply:
Yes, it's a good idea.

Reply:
Could be even better, don't you think?

<img width="1264" alt="Capture d’écran 2020-07-31 à 17 03 18" src="https://user-images.githubusercontent.com/8329789/89048163-c0475980-d34f-11ea-91c5-a09906eb8bc4.png">

Reply:
I added a commit to use this output, cause I think it's far better than my first proposal.

Reply:
I edited the first post with this output too.

Reply:
table is better imo yes, but depending on the amount of constrains

Reply:
Yes, let's wait for community to react with that proposal and maybe something better.

Reply:
See https://github.com/symfony/symfony/pull/37706#discussion_r469324212

I think using the VarDumper component is the best option here if available. Otherwise, the `Dumper` class gracefully handles this kind of output as well.

Reply:
Yes I think it's the best options! Moreover, there is less code to output the options results.